pkg/apis/internal/common.go

```diff
@@ -0,0 +1,39 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// CommonMethods has methods overridden for all GS objects in the same way
+type CommonMethods struct{}
+
+// FilterAPIResponse replaces the API response with an empty json object
+func (cm *CommonMethods) FilterAPIResponse(ctx context.Context, res *http.Response) (*http.Response, error) {
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if op == types.OperationDestroy {
+		err = res.Body.Close()
+		if err != nil {
+			return nil, err
+		}
+
+		res.Body = io.NopCloser(bytes.NewReader([]byte("{}")))
+		return res, nil
+	}
+	return res, nil
+}
+
+// State describes the current State of a GS object
+type State struct {
```

Comment:
This is the same type as `lbaasv1.State`, right? That one also already implements the fun of "sending to the server has another format than retrieving".

Sadly the actual state values vary by API, so we cannot fully extract that stuff .. I'd love to see this implementing `lbaasv1.StateRetriever` (and moving that interface out of lbaasv1) in some way, maybe API-specific `State` types that embed this generic one but add the methods or a way to retrieve the known state values?